src/compiler/crystal/syntax/parser.cr

```diff
@@ -3687,10 +3688,12 @@ module Crystal
       node
     end
 
-    def preserve_stop_on_do(new_value = false)
+    def preserve_stop_on_do(new_value = false, call_args? = true)
       old_stop_on_do = @stop_on_do
       @stop_on_do = new_value
+      @call_args_nest += 1 if call_args?
```

Comment:
Maybe call_args_nest should just be incremented and decremented in the methods `parse_call_args` and `parse_call_args_space_consumed`? I don't think this is the right place to do it.